### ISSUE TITLE
Add support for validating positional count in class patterns

### ISSUE THREAD

**Issue by aDotInTheVoid** [microsoft/pyright#6019]
**Describe the bug**
The following code has a type error, and pyright should report it.

**Code or Screenshots**

```python
# pyright: strict
from dataclasses import dataclass


@dataclass
class Alt:
    items: list[int]


def optimze(g: Alt) -> bool:
    match g:
        case Alt([_], 1):
            return True
        case _:
            return True


optimze(Alt([]))
```

Pyright thinks this code typechecks without problems:

```
$ ./node_modules/.bin/pyright ./bug.py 
0 errors, 0 warnings, 0 informations 
```

However this isn't true:

```
$ python3 ./bug.py 
Traceback (most recent call last):
  File "/home/alona/tmp/pbug/./bug.py", line 18, in <module>
    optimze(Alt([]))
  File "/home/alona/tmp/pbug/./bug.py", line 12, in optimze
    case Alt([_], 1):
         ^^^^^^^^^^^
TypeError: Alt() accepts 1 positional sub-pattern (2 given)
```

**VS Code extension or command-line**

Pyright CLI, `pyright 1.1.328`, installed with NPM. Python 3.11.4

**Comment by erictraut** [comment]
Thanks for the feature request. This functionality will be included in the next release.
